File: src/main/java/com/vjiki/music/entity/Playlist.java
package com.vjiki.music.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.OffsetDateTime;
import java.util.UUID;

@Entity
@Table(name = "playlists", schema = "music")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Playlist {

    @Id
    @GeneratedValue
    @Column(name = "id")
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(name = "name", nullable = false)
    private String name;

    @Column(name = "description", columnDefinition = "text")
    private String description;

    @Column(name = "cover_url", columnDefinition = "text")
    private String coverUrl;

    @Column(name = "type", nullable = false)
    @Builder.Default
    private String type = "CUSTOM";

    @Column(name = "is_public", nullable = false)
    @Builder.Default
    private Boolean isPublic = false;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private OffsetDateTime createdAt;

    @Column(name = "created_by", nullable = false, updatable = false)
    @Builder.Default
    private String createdBy = "system";

    @UpdateTimestamp
    @Column(name = "modified_at", nullable = false)
    private OffsetDateTime modifiedAt;

    @Column(name = "modified_by", nullable = false)
    @Builder.Default
    private String modifiedBy = "system";

    @Version
    @Column(name = "version", nullable = false)
    @Builder.Default
    private Integer version = 0;
}

File: src/main/kotlin/com/vjiki/music/entity/Playlist.kt
package com.vjiki.music.entity

import jakarta.persistence.*
import org.hibernate.annotations.CreationTimestamp
import org.hibernate.annotations.UpdateTimestamp
import java.time.OffsetDateTime
import java.util.*

@Entity
@Table(name = "playlists", schema = "music")
data class Playlist(
    @Id
    @GeneratedValue
    @Column(name = "id")
    val id: UUID = UUID.randomUUID(),

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    val user: User,

    @Column(name = "name", nullable = false)
    val name: String,

    @Column(name = "description", columnDefinition = "text")
    val description: String? = null,

    @Column(name = "cover_url", columnDefinition = "text")
    val coverUrl: String? = null,

    @Column(name = "type", nullable = false)
    val type: String = "CUSTOM",

    @Column(name = "is_public", nullable = false)
    val isPublic: Boolean = false,

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    val createdAt: OffsetDateTime? = null,

    @Column(name = "created_by", nullable = false, updatable = false)
    val createdBy: String = "system",

    @UpdateTimestamp
    @Column(name = "modified_at", nullable = false)
    var modifiedAt: OffsetDateTime? = null,

    @Column(name = "modified_by", nullable = false)
    var modifiedBy: String = "system",

    @Version
    @Column(name = "version", nullable = false)
    var version: Int = 0
)

